src/dos/dos_files.cpp

```diff
@@ -815,7 +815,23 @@ bool DOS_Canonicalize(const char* const name, char* const big)
 	return true;
 }
 
-bool DOS_GetFreeDiskSpace(uint8_t drive,uint16_t * bytes,uint8_t * sectors,uint16_t * clusters,uint16_t * free) {
+std::string DOS_Canonicalize(const char* const name)
+{
+	uint8_t drive;
+	char fullname[DOS_PATHLENGTH];
+	if (!DOS_MakeName(name, fullname, &drive)) {
+		return {};
+	}
+	std::string ret;
+	ret.push_back(drive + 'A');
```

Comment:
There's a support.h function:

`char drive_letter(uint8_t index)`

It asserts that the drive index is legal before giving back the drive letter

Reply:
Using this in latest commit